src/main/java/org/icatproject/core/entity/DataPublicationDate.java

```diff
@@ -0,0 +1,58 @@
+package org.icatproject.core.entity;
+
+import java.io.Serializable;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+@Comment("A date relevant for the publication")
+@SuppressWarnings("serial")
+@Entity
+@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "DATAPUBLICATION_ID", "DATETYPE" }) })
+public class DataPublicationDate extends EntityBaseBean implements Serializable {
+
+	@JoinColumn(name = "DATAPUBLICATION_ID", nullable = false)
+	@ManyToOne(fetch = FetchType.LAZY)
+	private DataPublication publication;
+
+	@Comment("Type of the date, see DataCite property dateType for suggested values")
+	@Column(name = "DATETYPE", nullable = false)
+	private String dateType;
+
+	@Comment("Use ISO 8601 format, may also be a range")
+	@Column(nullable = false)
```

Comment:
To make this work on Oracle where `DATE` is a reserved word, we are going to need to add something like `name = "DATE_"` to the definition of this column. I've tested this on MariaDB and Oracle and it works. The `upgrade_mysql/oracle_5_0.sql` scripts will need the column name updating as well.

Reply:
@RKrahl If you are happy with the re-naming of the data column to DATE_ then I'm happy to add this to my pull request along with the change required for the constraint names.